awx/main/tasks/receptor.py

```diff
@@ -411,9 +413,16 @@ def _run_internal(self, receptor_ctl):
                     return
 
                 try:
-                    resultsock = receptor_ctl.get_work_results(self.unit_id, return_sockfile=True)
-                    lines = resultsock.readlines()
-                    receptor_output = b"".join(lines).decode()
+                    receptor_output = ''
+                    if state_name == 'Failed' and self.task.runner_callback.event_ct == 0:
+                        # if receptor work unit failed and no events were emitted, work results may
+                        # contain useful information about why the job failed. In case stdout is
+                        # massive, only ask for last 1000 bytes
+                        startpos = max(stdout_size - 1000, 0)
+                        resultsock, resultfile = receptor_ctl.get_work_results(self.unit_id, startpos=startpos, return_socket=True, return_sockfile=True)
```

Comment:
What's are the version compatibility issues with this? Does this requires an updated receptor, and does that make it unreasonable to backport? Is there any idea to work on a different fix for backports?

Reply:
the receptor change is minimal, so backporting shouldn't be too hard. 

a more minimal version of this that doesn't require a receptor change is to not do get_work_results at all if stdout size > 1000.

Reply:
I see, so we could merge that and develop that solution you mention and test it separately. Fine by me.